Kubernetes/windows/debug/ValidateKubernetesHelper.psm1

```diff
@@ -0,0 +1,127 @@
+function GetContainers
+{
+    param(
+        [ValidateSet("Local", "Remote")][string] $PodLocation
+    )
+
+    $hostname = Get-WmiObject Win32_ComputerSystem | Select-Object -ExpandProperty name
+
+    $out = ((kubectl get pods -o json) | ConvertFrom-Json).items
+
+    $containers = @()
+    foreach ($c in $out)
+    {
+        $container = @{
+            Name = $c.metadata.name;
+            Service = $c.metadata.labels.app;
+            IPAddress = $c.status.podIP;
+            HostName = $c.spec.nodeName;
+            HostIP = $c.status.hostIP;
+            Status = $c.status.phase
+        }
+
+        #TODO: check for service name
+
+        if ($PodLocation -ieq "Local") {
+            if (!($container.HostName -ieq $hostname)) {
+                continue
+            }
+        }
+        if ($PodLocation -ieq "Remote") {
+            if ($container.HostName -ieq $hostname) {
+                continue
+            }
+        }
+        $containers += $container
+    }
+
+    return $containers;
+}
+
+function TestConnectivity()
+{
+    param(
+        [string] $containerName,
+        [string] $remoteHost =  "172.217.3.206", #google.com
+        [string] $port = "80",
+        [switch] $fromHost
+    )
+    if ($fromHost) {
+        $status = curl ${remoteHost}:${port} -UseBasicParsing -DisableKeepAlive
+        if ($status.StatusCode -eq "200") {
+            return
+        }
+        throw "TCP connection to ${remoteHost}:${port} failed from host"
+    } else {
+        $status = kubectl exec $containerName -- powershell.exe curl ${remoteHost}:${port} -UseBasicParsing -DisableKeepAlive
+    }
+
+    if ($status -match "200")
+    {
+        return
+    }
+
+    throw "TCP connection to ${remoteHost}:${port} failed from $containerName."
+}
+
+function GetContainerIPv4Address()
+{
+    param(
+        [string] $containerName
+    )
+
+    $matches = (kubectl exec $containerName ipconfig | Out-String  | Select-String -Pattern '(?sm)(IPv4 Address).*?: (.*?)\r\n' -AllMatches).Matches

```

Comment:
can you also add a test to validate if kubectl get pod <podname> is reporting the same podip as the container ip?

Reply:
This test compares ipconfig with kubectl reported ip